### PR TITLE
add apollo client

### LINKED ISSUE #6: [Client] Integrate Apollo Client into the Client
install the `apollo-boost` package, and add the apollo provider to the app root.

### PR DESCRIPTION
**Pull request by Tyresius92**
Closes #6 

This PR adds the apollo client to our client.

### CODE REVIEW

File: client/src/apolloClient.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import ApolloClient from 'apollo-boost';
```
Comment (marcgaj) [inline client/src/apolloClient.js:1]: Any reservations about changing this down the line? Would be nice to set up some push/pull style subscriptions with web sockets once the backend is further developed and I'm struggling to find documentation on how to configure different link types with this module

File: client/src/apolloClient.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+const client = new ApolloClient({
+  uri: 'http://localhost:4000',
+});
+
```
Comment (marcgaj) [inline client/src/apolloClient.js:6]: If you wanted to TDD new features here you could hot load the cache with mock data with writeQuery and completely sidestep the server development until it's all working e2e

**Comment by marcgaj** [comment]
Overall LGTM but left some food for thought comments

File: client/src/apolloClient.js (hunk 1 above)
Comment (Tyresius92) [reply client/src/apolloClient.js:1]: yeah apollo boost is the basic basic "idk if i need any fancy features" package. Once we hit a road block where we need to be more explicit we pop that sucker on. They've got a migration guide for when we hit that point.

File: client/src/apolloClient.js (hunk 2 above)
Comment (Tyresius92) [reply client/src/apolloClient.js:6]: Yeah we can definitely do some writeQuery stuff if you're interested in starting client side development. I've been enjoying the server side stuff at the moment, so was focusing on that. 

But yeah, feel free to create some issues and spin up some PRs.

**Pull request merged**